website/components/LandingPage/OpenSource/index.js

```diff
@@ -37,43 +37,45 @@ function cacheKey(key) {
 function cacheGet(key) {
   try {
     const { expires, data } = JSON.parse(localStorage.getItem(cacheKey(key)));
-    if (expires > Date.now()) return data;
+    return { expired: expires < Date.now(), data };
   } catch (error) {
     // Nothing.
   }
+  return {};
 }
 
-function cacheSet(key, data, length = 120000) {
-  const expires = Date.now() + length;
+function cacheSet(key, data, expires) {
   try {
     localStorage.setItem(cacheKey(key), JSON.stringify({ data, expires }));
   } catch (error) {
     // Nothing.
   }
 }
 
-function cached(key, fallback, length) {
-  const data = cacheGet(key);
-  if (data) return Promise.resolve(data);
+function cached(key, fallback, expires) {
+  const { expired, data } = cacheGet(key);
 
-  return fallback().then(data => {
-    cacheSet(key, data, length);
-    return data;
-  }, console.error);
+  if (data === undefined || expired === true) {
+    return fallback()
+      .then(data => {
+        cacheSet(key, data, expires);
+        return data;
+      })
+      .catch(() => data || 0);
+  }
+
+  return Promise.resolve(data);
 }
 
 function formatDate(date) {
-  return date.toISOString().slice(0, 10);
+  return date && date.toISOString().slice(0, 10);
```

Comment:
`''`, 'undefined`, `false` or `null` is a very bad result in here. Why is this check actually needed?

Reply:
Github api after rate limit has been surpassed returns empty object `{}` as a response, resulting in `stars` and `forks` being `undefined`. I will move this check somewhere else.